File: backend/src/main/java/tech/rahulpandey/backend/model/Event.java
package tech.rahulpandey.backend.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.*;

import java.util.Date;
import java.util.List;

@Entity
public class Event {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(unique = true)
    private String name;

    private String slug;

    private String description;

    private List<String> rules;

    private EventCategory category;

    private List<String> prizes;

    @JsonFormat(pattern = "dd-mm-yyyy")
    private Date registrationDeadline;

    private String teamSize;

    private List<String> eventDates;

    @ElementCollection
    private List<Organiser> organisers;

    @OneToOne
    private Users user;

}

File: frontend/src/types/Event.ts
type EventCategory =
  | "CODING"
  | "CIRCUITS_AND_ROBOTICS"
  | "BUSINESS"
  | "BRAINSTORMING"
  | "GAMING"
  | "MISCELLANEOUS";

type Organiser = {
    name: string;
    phoneNumber: string;
}

type StringObj = {
  value: string;
}

type EventFormat = {
  name: string;
  slug: string;
  description: string;
  rules: string[];
  category: EventCategory;
  prizes: string[];
  registrationDeadline: string;
  teamSize: string;
  eventDates: string[];
  organisers: Organiser[];
}

type EventFormType = {
  name: string;
  slug: string;
  description: string;
  rules: StringObj[];
  category: EventCategory;
  prizes: StringObj[];
  registrationDeadline: string;
  teamSize: string;
  eventDates: StringObj[];
  organisers: Organiser[];
}

export type { EventFormat, EventCategory, EventFormType };
